pytact/data_reader.py

```diff
@@ -568,53 +568,43 @@ def get_index(index):
             return cast(Definition, Node(depgraph, lreader[depgraph].representative, lreader).definition)
         return TupleLike(count, get_index)
 
-    def _global_context(self, seen: set[Node]) -> Iterable[Definition]:
-        if prev := self.previous:
-            yield prev
-            yield from prev._global_context(seen)
-        for eprev in self.external_previous:
-            if eprev.node not in seen:
-                yield eprev
-                yield from eprev._global_context(seen)
-                seen.add(eprev.node)
-
-    def _clustered_global_context(self, seen: set[Node]) -> Iterable[list[Definition]]:
-        if prev := self.previous:
-            cluster = list(prev.cluster)
-            yield cluster
-            yield from cluster[-1]._clustered_global_context(seen)
-        for eprev in self.external_previous:
-            if eprev.node not in seen:
-                cluster = list(eprev.cluster)
-                yield cluster
-                yield from cluster[-1]._clustered_global_context(seen)
-                seen.add(eprev.node)
-
-    @property
-    def global_context(self) -> Iterable[Definition]:
+    def global_context(self, across_files : bool = True, inclusive = False, seen : set[Node] = None) -> Iterable[Definition]:
         """All of the definitions in the global context when this definition was created.
 
         Note that this does not include this definition itself, except when the definition is a inductive,
         constructor or projection. Because those are mutually recursive objects, they reference themselves
         and are therefore part of their own global context.
```

Comment:
You should probably add some documentation for `across_files`, `inclusive` and `set`. Also in the other functions that you modified with these parameters.